weldx/core.py

```diff
@@ -708,3 +711,557 @@ def units(self) -> pint.Unit:
+        if isinstance(self._obj, xr.DataArray):
+            return self._obj.__getitem__(*args)
+        return NotImplementedError
+
+    def interp_like(
+        self, obj: Any, dimensions: List[str] = None, accessor_mappings: Dict = None
+    ) -> GenericSeries:
+        """Interpolate using the coordinates of another object.
+
+        Parameters
+        ----------
+        obj :
+            An object that provides the coordinate values.
+        dimensions :
+            The dimensions that should be interpolated. If `None` is passed, all
+            dimensions will be interpolated
+        accessor_mappings :
+            A mapping between the dimensions of the generic series and the corresponding
+            coordinate accessor of the provided object. This can be used if the
+            coordinate names do not match for the time series and the provided object.
+
+        Returns
+        -------
+        GenericSeries :
+            A new generic series containing discrete values for the interpolated
+            dimensions.
+
+        """
+        raise NotImplementedError
+
+    @property
+    def coordinates(self) -> Union[None, pint.Quantity, Dict[str, pint.Quantity]]:
+        """Get the coordinates of the generic series."""
+        if isinstance(self._obj, xr.DataArray):
+            return self._obj.coords
+        # todo here we should get all parameter coordinates
+        return None
+
+    @property
+    def coordinate_names(self) -> List[str]:
+        """Get the names of all coordinates."""
+        raise NotImplementedError
+
+    @property
+    def data(self) -> Union[pint.Quantity, MathematicalExpression]:
+        """Get the internal data."""
+        if isinstance(self._obj, xr.DataArray):
+            return self._obj.data
+        return self._obj
+
+    @property
+    def data_array(self) -> Union[xarray.DataArray, MathematicalExpression]:
+        """Get the internal data."""
+        if isinstance(self._obj, xr.DataArray):
+            return self._obj
+        return None
+
+    @staticmethod
+    def _get_expression_dims(expr: MathematicalExpression, symbol_dims: Dict[str, str]):
+        dims = set()
+        for d in symbol_dims.values():
+            dims |= set(d)
+        for v in expr.parameters.values():
+            if not isinstance(v, xr.DataArray):
+                v = xr.DataArray(v)
+            if v.size > 0:
+                dims |= set(v.dims)
+        return list(dims)
+
+    @property
+    def dims(self) -> List[str]:
+        """Get the names of all dimensions."""
+        if isinstance(self._obj, MathematicalExpression):
+            return self._get_expression_dims(self._obj, self._symbol_dims)
+        return self._obj.dims
+
+    @property
+    def interpolation(self) -> str:
+        """Get the name of the used interpolation method."""
+        return self._interpolation
+
+    @interpolation.setter
+    def interpolation(self, val: str):
+        if val not in ["linear", "step"]:
```

Comment:
we should double check interpolation methods available with array (didn't we define them somewhere already?)

it probably makes sense to distinguish between 1d/nd cases